### PR TITLE
fix(`forge`): pre-emptively create `lib` dir if it doesn't exist for updating submodules

### LINKED ISSUE #6519: Running "forge install" in a project with no submodule dependencies does not work
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (d4e6b43 2023-12-04T00:30:25.537736000Z)

### What command(s) is the bug in?

forge install

### Operating System

macOS (Apple Silicon)

### Describe the bug

Running `forge install` in a project with no submodule dependencies does not work. Forge will produce the following error:

```text
Updating dependencies in /Users/prb/Projects/prb-proxy/lib
error: pathspec '/Users/prb/Projects/prb-proxy/lib' did not match any file(s) known to git
Error:
git submodule exited with code 1
```

It looks like Foundry assumes that there is a `lib` directory and that there are some submodules in there.

You can quickly reproduce the bug using [PRBProxy](https://github.com/PaulRBerg/prb-proxy) and the instructions provided here:

https://github.com/crytic/slither-action/issues/44#issuecomment-1838832654

This is a blocker so a fix would be appreciated 🙏 

Cc @DaniPopes @mattsse

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

Closes #6519 

## Solution

Check if there are any submodules at all in the repo, and only update if so.

### CODE REVIEW

File: crates/forge/bin/cmd/install.rs [hunk 1]
```diff
@@ -123,12 +123,33 @@ impl DependencyInstallOpts {
...
-            git.submodule_update(false, false, false, true, Some(&libs))?;
+
+            let empty_install_dir = libs.read_dir()?.next().is_none();
+
+            if empty_install_dir {
+                fs::create_file(libs.join(".keep"))?;
+                // Update the git project with the newly created folder if we're using git.
+                if !self.no_git {
+                    git.add(Some(&libs.join(".keep")))?;
+                }
```
Comment (mattsse) [inline crates/forge/bin/cmd/install.rs:139]: why do we need to handle a .keep file?

Comment (Evalir) [reply crates/forge/bin/cmd/install.rs:139]: Git won't track empty folders, so we need a keep file so we can track it.

Comment (DaniPopes) [reply crates/forge/bin/cmd/install.rs:139]: I don't think this is right, what is the point of running `git submodule update` on an empty directory?

Comment (Evalir) [reply crates/forge/bin/cmd/install.rs:139]: Yeah, we can probably just skip alltogether if the dir is empty.

**Review by mattsse — COMMENTED** [review]
the failing test look related

**Comment by Evalir** [comment]
Yeah so what's happening here is an edge case for `prb-proxy`: The reason why we wanna be able to run `submodule update` on an empty `lib` file is that we could have missing deps that might have been removed (manually), and we need to reinstall them through a `.gitmodules`. `prob-proxy` doesn't have one, so it will just not install any deps at all. 

I think the fix here is to check if the `.gitmodules` file exists, and if not, skip the submodule update. If it exists, even if the dir is empty we wanna run the update.

File: crates/forge/bin/cmd/install.rs [hunk 2]
```diff
@@ -123,12 +123,31 @@ impl DependencyInstallOpts {
...
             p_println!(!self.quiet => "Updating dependencies in {}", libs.display());
-            // recursively fetch all submodules (without fetching latest)
-            git.submodule_update(false, false, false, true, Some(&libs))?;
+
+            let empty_install_dir = libs.read_dir()?.next().is_none();
+
+            if !empty_install_dir {
+                fs::create_file(libs.join(".keep"))?;
+                // Update the git project with the newly created folder if we're using git.
+                git.add(Some(&libs.join(".keep")))?;
```
Comment (mattsse) [inline crates/forge/bin/cmd/install.rs:139]: this looks super hacky,

I'm still skeptical that any of this is necessary.

The edge you're described is correct, but this can only happen if the lib dir is manually removed but already known to git.

so what we really want to know here is "does this repo contain any submodules"
I think the right fix here is querying `git submodule status` first
if this doesn't give us submodules, then there's nothing to update

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/forge/bin/cmd/install.rs (hunk 2 above)
Comment (Evalir) [reply crates/forge/bin/cmd/install.rs]: impl'd in https://github.com/foundry-rs/foundry/pull/6521/commits/84a83a95fd507a722f0446f97c892ec09a5d2a9e

**Review by mattsse — CHANGES_REQUESTED** [review]
can we add a sanity test for this:

empty project + call `forge install`

**Review by mattsse — APPROVED** [review]

**Comment by PaulRBerg** [comment]
Thanks for the quick turnaround 🙏

**Comment by PaulRBerg** [comment]
There's still a problem left, though .. running `forge install` will add the following setting in `foundry.toml`:

```toml
libs = ["node_modules", "lib"]
```

Would it be possible to have Forge not do this?
